examples/cpp/quick-start.cpp

```diff
@@ -0,0 +1,173 @@
+                std::cout << "The object was deleted.\n";
+            } else {
+                for (auto& propertyChange : change.property_changes) {
+                    std::cout << "The object's " << propertyChange.name << " property has changed.\n";
+                    CHECK(propertyChange.name == "status"); // :remove:
+                }
+            }
+        } catch (std::exception const& e) {
+            std::cerr << "Error: " << e.what() << "\n";
+        }
+    });
+    // :snippet-end:
+
+    // :snippet-start: modify-write-block
+    auto todoToUpdate = todosInProgress[0];
+    realm.write([&realm, &todoToUpdate] {
+        todoToUpdate.status = "Complete";
+    });
+    // :snippet-end:
+    CHECK(*todoToUpdate.status == "Complete");
+
+    // :snippet-start: delete
+    realm.write([&realm, &todo] {
+        realm.remove(todo);
+    });
+    // :snippet-end:
+}
+
+TEST_CASE("sync quick start", "[realm][write][sync]") {
+    // :snippet-start: connect-to-backend
+    auto app = realm::App(APP_ID);
+    // :snippet-end:
+    
+    // :snippet-start: authenticate-user
+    auto user = app.login(realm::App::credentials::anonymous()).get_future().get();
+    // :snippet-end:
+    
+    // :snippet-start: open-synced-realm
+    auto sync_config = user.flexible_sync_configuration();
+    auto synced_realm_ref = realm::async_open<Sync_Todo>(sync_config).get_future().get();
+    auto realm = synced_realm_ref.resolve();
+    // :remove-start:
+    // Remove any existing subscriptions before adding the one for this example
+    auto clearInitialSubscriptions = realm.subscriptions().update([](realm::mutable_sync_subscription_set &subs) {
+        subs.clear();
+    }).get_future().get();
+    CHECK(clearInitialSubscriptions == true);
+    CHECK(realm.subscriptions().size() == 0);
+    // :remove-end:
+    // For this example, get the userId for the Flexible Sync query
+    auto userId = user.identifier();
+    auto subscriptions = realm.subscriptions();
+    auto updateSubscriptionSuccess = subscriptions.update([&](realm::mutable_sync_subscription_set &subs) {
+        subs.add<Sync_Todo>("todos", [&userId](auto &obj) {
+            // For this example, get only Sync_Todo items where the ownerId
+            // property value is equal to the userId of the logged-in user.
+            return obj.ownerId == userId;
+        });
+    }).get_future().get();
+    // :snippet-end:
+    CHECK(updateSubscriptionSuccess == true);
+    
+    // The C++ SDK is currently missing a constructor to store a std::string
+    // So convert the userId std::string to a character array for persisting.
+    // TODO: Remove this and use the userId directly when the constructor is added.
+    char* charUserId = strcpy(new char[userId.length() + 1], userId.c_str());
+    
+    // :snippet-start: write-to-synced-realm
+    auto todo = Sync_Todo {
+        .name = "Create a Sync todo item",
+        .status = "In Progress",
+        .ownerId = charUserId
```

Comment:
This doesn't work? (Workaround either way)
```suggestion
        .ownerId = userId.c_str()
```

Reply:
Ah, that works! Thank you!